java/sage/Seeker.java

```diff
@@ -1241,6 +1241,17 @@ else if (autoImportAddedNetPaths.size() > 0)
       writeArchiveDirsProps();
     }
 
+    // Even though this is the only method that manipulates this variable, it is run on a thread
```

Comment:
Now I'm not sure why I thought that would do anything. Must has been a late night. The better way to make that thread safe would be to make autoImportAddedNetPaths volatile and replace it with a new Set each time if anything changed.